peer-mgr/peerMgr.go

```diff
@@ -0,0 +1,53 @@
+package peer_mgr
+
+import (
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/meshplus/bitxhub-model/pb"
+)
+
+type BasicPeerManager interface {
+	// Start
+	Start() error
+
+	// Stop
+	Stop() error
+
+	// AsyncSend sends message to peer with peer info.
+	AsyncSend(KeyType, *pb.Message) error
+
+	// Send sends message waiting response
+	Send(KeyType, *pb.Message) (*pb.Message, error)
+
+	// CountConnectedPeers counts connected peer numbers
+	CountConnectedPeers() uint64
+}
+
+//go:generate mockgen -destination mock_orderPeermgr/mock_orderPeermgr.go -package mock_orderPeermgr -source peermgr.go
+type OrderPeerManager interface {
+	BasicPeerManager
+
+	// SubscribeOrderMessage
+	SubscribeOrderMessage(ch chan<- OrderMessageEvent) event.Subscription
+
+	// AddNode adds a vp peer.
+	AddNode(newNodeID uint64, vpInfo *pb.VpInfo)
+
+	// DelNode deletes a vp peer.
+	DelNode(delID uint64)
+
+	// UpdateRouter update the local router to quorum router.
+	UpdateRouter(vpInfos map[uint64]*pb.VpInfo, isNew bool) bool
+
+	// OtherPeers return peers except local peer.
+	OtherPeers() map[uint64]*peer.AddrInfo
```

Comment:
OtherPeers and Broadcast can put in BasicPeerManager

Reply:
consensus node will use these function, can not move in the current design.